api/src/main/java/org/apache/iceberg/io/FileIO.java

```diff
@@ -29,7 +29,7 @@
  * must be serializable because various clients of Spark tables may initialize this once and pass
  * it off to a separate module that would then interact with the streams.
  */
-public interface FileIO extends Serializable {
+public interface FileIO extends Serializable, CatalogConfigurable {
```

Comment:
I'm not sure it actually makes sense for the FileIO to implement `CatalogConfigurable`.  For S3FileIO, the initialization is just setting up the `AwsProperties`, which is also being set as part of the constructor and those two paths actually conflict (as you note in the comments).  However, this makes S3FileIO pretty confusing. (I'll add more comments on S3FileIO).

Reply:
In that case, let me remove the `CatalogConfigurable` related changes first. I do think it is worth adding, but seems too much to discuss in this PR.

Reply:
I just want to be sure that we understand how the properties are getting propagated to the FileIO.  Right now, it seems that we're loading it via the CatalogUtils, but only in the glue catalog, so it seems somewhat specific at this point.  Since the glue catalog can just load the properties and use the S3FileIO constructor, it seems like we should pull it out from the PR and revisit with a broader change address the default dynamic loading.

I agree that it might be a good addition because if you dynamically load FileIO right now, I don't believe you can set custom AwsProperties.